aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java

```diff
@@ -709,21 +713,68 @@ private void launchSignOut(final String redirectUri, final String browserPackage
         launchCustomTabs(fqdn, null, browserPackage);
     }
 
+    /***
+     * Check if a browser is installed on the device to launch HostedUI.
+     * @return true if a browser exists else false.
+     */
+    public Boolean isBrowserInstalled() {
+        String url = "https://docs.amplify.aws/";
+        Uri webAddress = Uri.parse(url);
+        Intent intentWeb = new Intent(Intent.ACTION_VIEW, webAddress);
+        return (intentWeb.resolveActivity(context.getPackageManager()) != null);
+    }
+
+    /**
+     * Get list of browser packages that support Custom Tabs Service.
+     * @return list of package names that support Custom Tabs.
+     */
+    private ArrayList<String> getSupportedBrowserPackage(){
```

Comment:
It is usually best to return a more generic type such as `List<String` or `Collection<String>` depending on how the method is being used. Since we are only using the result of this method to check for the size, `Collection<String>` would make most sense here.

Reply:
Actually, why cant this method simply be `isCustomTabsSupported()` and return a `boolean`? Are we using the list of supported packages elsewhere?

Reply:
I have wrapped `getSupportedBrowserPackage` into `isCustomTabsSupported()`. Although we are not using the list, kept it just to be more readable.